src/database/cloudant.js

```diff
@@ -29,11 +29,11 @@ class CloudantDatabase {
           if (insertId) {
             documentWithId.id = id;
           }
-          database.insert(documentWithId, (err) => {
+          database.insert(documentWithId, (err, result) => {
             if (err) {
               logger.error(`Error occurred: ${err.message} create()`);
               reject(err);
-            } else {
+            } else if (result.ok === true) {
               resolve({ data: documentWithId, statusCode: 200 });
             }
```

Comment:
Would be good to have a `reject("Error: object cannot be created")` inside an "else" statement

Reply:
you have to return the error with reject